src/zcash/Note.cpp

```diff
@@ -0,0 +1,92 @@
+#include "Note.hpp"
+#include "prf.h"
+#include "crypto/sha256.h"
+#include "zerocash/utils/util.h"
+
+#include "version.h"
+#include "streams.h"
+
+namespace libzcash {
+
+Note::Note() {
+    a_pk = random_uint256();
+    rho = random_uint256();
+    r = random_uint256();
+    value = 0;
+}
+
+uint256 Note::cm() const {
+    unsigned char discriminant = 0xb0;
+
+    CSHA256 hasher;
+    hasher.Write(&discriminant, 1);
+    hasher.Write(a_pk.begin(), 32);
+
+    std::vector<unsigned char> value_vec(sizeof(value), 0);
+    libzerocash::convertIntToBytesVector(value, value_vec);
```

Comment:
Hm, yes this should be changed to match the spec. I retained the same behavior and encoding from the old implementation.